src/components/ResourceList/components/FilterControl/components/DateSelector/tests/DateSelector.test.tsx

```diff
@@ -161,6 +161,32 @@ describe('<DateSelector />', () => {
     });
   });
 
+  describe('timezones adjustments', () => {
+    const timeZoneOffsets = [720, 540, 60, 0, -60, -540, -720];
+    const date = '2019-01-01';
+    const getTimezoneOffset = Date.prototype.getTimezoneOffset;
+
+    timeZoneOffsets.forEach((zone) => {
+      it(`sets the correct values for ${date} with timezone offset ${zone}`, () => {
+        Date.prototype.getTimezoneOffset = () => zone; // eslint-disable-line no-extend-native
```

Comment:
Is this allowed? Can't you use `mockImplementation`? it feels a bit weird to overwrite it like that.